force-app/main/educationCloudSettings/lwc/edcSettingsProductDisplay/edcSettingsProductDisplay.js

```diff
@@ -0,0 +1,48 @@
+import { LightningElement, api, track } from 'lwc';
+import stgHealthCheckLoadingIndicator from "@salesforce/label/c.stgHealthCheckLoadingIndicator";
+import { ShowToastEvent } from "lightning/platformShowToastEvent";
+
+export default class EdcSettingsProductDisplay extends LightningElement {
+    @api productRegistryVModels;
+
+    @track loadedSettingsProductCount = 0;
+    @track displaySettingsProduct = false;
+    errorList = [];
+
+    LabelReference = {
+        spinnerLoadingAltText: stgHealthCheckLoadingIndicator,
+    };
+
+    handleSettingsProductLoaded(evt) {
+        this.loadedSettingsProductCount++;
+        this.areAllProductsLoaded();
+    }
+
+    handleSettingsProductError(evt){
+        const errorMessage = evt.detail.errorMessage;
+        this.errorList.push(errorMessage);
+        this.loadedSettingsProductCount++;
+        this.areAllProductsLoaded();
+    }
+
+    areAllProductsLoaded(){
+        if (this.loadedSettingsProductCount == this.productRegistryVModels.length) {
+            this.displaySettingsProduct = true;
+            if (this.errorList && this.errorList.length > 0){
+                this.errorList.forEach(error => {
+                    this.createShowToast(error);
+                });
+            }
+        }
```

Comment:
Consider changing this to a guard clause to force code coverage in tests and to make the logic more clear. Avoiding the pyramid of doom usually pays off.


```suggestion
        if (this.loadedSettingsProductCount !== this.productRegistryVModels.length) {
            return;
        }
        
        this.displaySettingsProduct = true;
        if (this.errorList && this.errorList.length > 0){
            this.errorList.forEach(error => {
                this.createShowToast(error);
            });
        }
```

Reply:
Done. Thanks!